portal/views/user.py

```diff
@@ -1359,6 +1359,77 @@ def user_documents(user_id):
     return jsonify(user_documents=[d.as_json() for d in
                                        user.documents])
 
+
+@user_api.route('/user/<int:user_id>/user_documents/<int:doc_id>/download')
```

Comment:
is there a need for the /download portion of the request path?  I'm not a RESTful design expert, but it seems like you typically just ask for the resource.  Is something else provided at /user/<user_id>/user_documents/<doc_id> ?